W_on_bags_Classes.py

```diff
@@ -0,0 +1,250 @@
+from typing import Union
+
+
+class ProductError(Exception):
+
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return self.message
+
+
+class Product:
+    def __init__(self, name: str, price: float) -> None:
+        """
+        конструктор класу Product
+        :param name: найменування продукту
+        :param price: вартість одиниці продукти
+        """
+        self.name = name
+        self.price = price
+
+    def validate(self) -> bool:
+        """
+        метод здійснює валідацію значень екземпляру класа
+        :return:
+        """
+        return True if self.name and self.price > 0 else False
+
+    def total_price(self, quantity: float) -> float:
+        """
+        метод повертає вартість продукту у кількості quantity
+        :param quantity: кількість продукту
+        :return: вартість вказаної кількості продукту
+        """
+        if self.validate():
+            return round(self.price * quantity, 2)
+        else:
+            raise ProductError('none validate object, product name {0}'.format(self.name))
+
+    def __eq__(self, other) -> bool:
+
+        return True if isinstance(other, Product) and self.name == other.name and self.price == other.price else False
```

Comment:
Доволі довгий та важкий до розуміня рядок.
Розділи поведінку.
```suggestion
        if not isinstance(other, Product):
            return False
        return self.name == other.name and self.price == other.price
```